### PR TITLE
Deprecate random.KeyArray and random.PRNGKeyArray

### PR DESCRIPTION
**Pull request by jakevdp**
Part of #9263

### CODE REVIEW

File: jax/random.py [hunk 1]
```diff
@@ -130,22 +130,12 @@
 https://jax.readthedocs.io/en/latest/notebooks/Distributed_arrays_and_automatic_parallelization.html#generating-random-numbers
 """
 
-from jax._src.prng import PRNGKeyArray as _PRNGKeyArray
 # TODO(frostig): remove this typechecking workaround. Our move away
```
Comment (froystig) [inline jax/random.py:133]: Remove this comment as well?

File: jax/experimental/jax2tf/tests/jax2tf_limitations.py [hunk 2]
```diff
@@ -1478,7 +1478,7 @@ def custom_assert(tst, result_jax, result_tf, *, args, tol, err_msg):
     # TODO(frostig): Don't need this conditional once we always
```
Comment (froystig) [inline jax/experimental/jax2tf/tests/jax2tf_limitations.py:1478]: We can remove this comment, or change it to one that says that we're handling both new- and old-style keys here.

**Review by froystig — APPROVED** [review]
So good!

File: jax/experimental/jax2tf/tests/jax2tf_limitations.py (hunk 2 above)
Comment (jakevdp) [reply jax/experimental/jax2tf/tests/jax2tf_limitations.py]: Done

File: jax/random.py (hunk 1 above)
Comment (jakevdp) [reply jax/random.py]: Done

**Pull request merged**